src/io/evented.cr

```diff
@@ -88,6 +88,23 @@ module IO::Evented
     end
   end
 
+  def evented_sendfile(remaining, errno_msg : String) : Nil
+    while remaining > 0
+      bytes_written = (yield remaining).to_i32
+      if bytes_written == -1
+        if Errno.value == Errno::EAGAIN
+          wait_writable
+        else
+          raise IO::Error.from_errno(errno_msg)
```

Comment:
Should be a `Socket::Error` here I think

Reply:
Ok, but would be nice to see what errno it is..

Reply:
`Socket::Error` contains `errno` too....